core/new-gui/src/app/workspace/service/validation/validation-workflow.service.ts

```diff
@@ -0,0 +1,141 @@
+   * @param texeraGraph
+   * @param workflowActionService
+   */
+  constructor(private operatorMetadataService: OperatorMetadataService,
+    private workflowActionService: WorkflowActionService) {
+    this.operatorMetadataService.getOperatorMetadata().subscribe(
+        value => { this.operatorSchemaList = value.operators; }
+      );
+
+    this.workflowActionService.getTexeraGraph().getOperatorAddStream()
+      .subscribe(
+        value => {
+          const operatorID = value.operatorID;
+          const status = this.validateOperator(operatorID);
+          this.operatorValidationStream.next({status, operatorID});
+          }
+        );
+
+    this.workflowActionService.getTexeraGraph().getOperatorDeleteStream()
+      .subscribe(
+        value => {
+          const operatorID = value.deletedOperator.operatorID;
+          const status = this.validateOperator(operatorID);
+          this.operatorValidationStream.next({status, operatorID});
+         }
+        );
+
+    this.workflowActionService.getTexeraGraph().getLinkAddStream()
+      .subscribe(value => {
+        let operatorID = value.source.operatorID;
+        let status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+
+    this.workflowActionService.getTexeraGraph().getLinkDeleteStream()
+      .subscribe(value => {
+        let operatorID = value.deletedLink.source.operatorID;
+        let status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+        operatorID = value.deletedLink.target.operatorID;
+        status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+
+    this.workflowActionService.getTexeraGraph().getOperatorPropertyChangeStream()
+      .subscribe(value => {
+        const operatorID = value.operator.operatorID;
+        const status = this.validateOperator(operatorID);
+        this.operatorValidationStream.next({status, operatorID});
+      });
+  }
+
+  /**
+   * Gets the observable for operator validation change event.
+   * Contains a boolean variable that indicates:
+   *  - the new status for the validation of operator
+   */
+  public getOperatorValidationStream(): Observable<{status: boolean, operatorID: string}> {
+    return this.operatorValidationStream.asObservable();
+  }
+
+  /**
+   * if all ports of the operator are connected and all properties are completed
+   * then the operator is valid.
+   */
+  public validateOperator(operatorID: string): boolean {
+    if (!this.isOperatorIsolated(operatorID) && this.isJsonSchemaValiad(operatorID)) {
+      return true;
+    }
+    return false;
```

Comment:
just write `return (!this.isOperatorIsolated(operatorID) && this.isJsonSchemaValiad(operatorID))`